File: src/2485-Find-the-Pivot-Integer/find_the_pivot_integer.js
/**
 * @param {number} n
 * @return {number}
 */
var pivotInteger = function(n) {
  let total_sum = (n * (n + 1)) / 2
  let left_sum = 0

  for (let i = 1; i <= n; i++) {
    left_sum += i
    let right_sum = total_sum - left_sum + i
    if (left_sum === right_sum) {
      return i
    }
  }

  return -1
};

let n = 8

console.log(pivotInteger(n))

File: src/2485-Find-the-Pivot-Integer/find_the_pivot_integer.py
class Solution:
    def pivotInteger(self, n: int) -> int:
        total_sum: int = (n * (n + 1)) // 2
        left_sum: int = 0 

        for i in range(1, n + 1):
            left_sum += i
            right_sum: int = total_sum - left_sum + i
            if left_sum == right_sum:
                return i
        
        return -1

if __name__ == "__main__":
    n: int = 8
    solution = Solution()

    print(solution.pivotInteger(n))